apps/nextjs/src/stores/chatStore/hooks.ts

```diff
@@ -1,17 +1,43 @@
 import { useEffect } from "react";
 
-import type { Message as AiMessage } from "ai";
+import type {
+  Message as AiMessage,
+  ChatRequestOptions,
+  CreateMessage,
+  Message,
+} from "ai";
 
 import { useChatStore } from "./index";
 
 // Hooks to update the Zustand chat store mirror from the AI SDK outputs
 export const useChatStoreMirror = (
   messages: AiMessage[],
   isLoading: boolean,
+  stop: () => void,
+  append: (
+    message: Message | CreateMessage,
+    chatRequestOptions?: ChatRequestOptions | undefined,
+  ) => Promise<string | null | undefined>,
+  reload: () => void,
 ) => {
   const setMessages = useChatStore((state) => state.setMessages);
+  const setStop = useChatStore((state) => state.setStop);
+  const setAppend = useChatStore((state) => state.setAppend);
+  const setReload = useChatStore((state) => state.setReload);
 
   useEffect(() => {
     setMessages(messages, isLoading);
   }, [messages, isLoading, setMessages]);
+
+  useEffect(() => {
+    setStop(stop);
+  }, [stop, setStop]);
+
+  useEffect(() => {
+    setAppend(append);
+  }, [append, setAppend]);
+
+  useEffect(() => {
+    setReload(reload);
+  }, [reload, setReload]);
```

Comment:
There are a couple of related points here which might make things simpler when you add them together:
- Are you sure that these functions are stable? When I tried this I found that they were updating more often than they should be, potentially on every render. I think we only need to set the initial value. In my branch I did this with `useEffect(..., [])`
- I think these actions can all be set in one call to the store, like `state.setAiSdkActions`?
- Should we consider putting the actions under a single key in the store, like actions.stop, actions.regenerate, etc?